attestation-service/verifier/src/snp/mod.rs

```diff
@@ -209,21 +220,33 @@ fn verify_signature(cert: &X509, issuer: &X509, name: &str) -> Result<()> {
         .ok_or_else(|| anyhow!("Invalid {name} signature"))
 }
 
-fn verify_cert_chain(cert_chain: &[CertTableEntry], ask: &X509, ark: &X509) -> Result<X509> {
-    let raw_vcek = cert_chain
+fn verify_cert_chain(
+    cert_chain: &[CertTableEntry],
+    ask: &X509,
+    ark: &X509,
+    asvk: &X509,
+) -> Result<X509> {
+    let raw_vek = cert_chain
         .iter()
-        .find(|c| c.cert_type == CertType::VCEK)
-        .ok_or_else(|| anyhow!("VCEK not found."))?;
-    let vcek = x509::X509::from_der(raw_vcek.data()).context("Failed to load VCEK")?;
-
-    // ARK -> ARK
-    verify_signature(ark, ark, "ARK")?;
-    // ARK -> ASK
-    verify_signature(ask, ark, "ASK")?;
-    // ASK -> VCEK
-    verify_signature(&vcek, ask, "VCEK")?;
-
-    Ok(vcek)
+        .find(|c| c.cert_type == CertType::VCEK || c.cert_type == CertType::VLEK)
+        .ok_or_else(|| anyhow!("VEK not found."))?;
+    let vek = x509::X509::from_der(raw_vek.data()).context("Failed to load VEK")?;
+
+    if raw_vek.cert_type == CertType::VCEK {
+        // Chain: ARK -> ARK -> ASK -> VCEK
+        verify_signature(ark, ark, "ARK")?;
+        verify_signature(ask, ark, "ASK")?;
+        verify_signature(&vek, ask, "VCEK")?;
+    } else if raw_vek.cert_type == CertType::VLEK {
+        // Chain: ARK -> ARK -> ASVK -> VLEK
+        verify_signature(ark, ark, "ARK")?;
+        verify_signature(asvk, ark, "ASVK")?;
+        verify_signature(&vek, asvk, "VLEK")?;
+    } else {
+        return Err(anyhow!("VEK not found."));
```

Comment:
Again, not sure VEK is the right term here. Also, this case is not that the cert isn't found, but that the type is not what we expect.